packages/ui-menu/src/Menu/index.tsx

```diff
@@ -231,87 +199,83 @@ class Menu extends Component<MenuProps> {
-    // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-    if (this._activeSubMenu && menuItem !== this._activeSubMenu._trigger) {
+  handleMenuItemMouseOver: MenuItemProps['onMouseOver'] = (event, menuItem) => {
+    if (this._activeSubMenu) {
       this.hideActiveSubMenu(event)
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  hideActiveSubMenu = (event) => {
+  hideActiveSubMenu = (event: React.MouseEvent | React.KeyboardEvent) => {
     if (this._activeSubMenu) {
-      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
       this._activeSubMenu.hide(event)
       this._activeSubMenu = null
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'shown' implicitly has an 'any' type.
-  handleSubMenuToggle = (shown, subMenu) => {
+  handleSubMenuToggle: MenuProps['onToggle'] = (shown, subMenu) => {
     if (shown) {
       this._activeSubMenu = subMenu
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  handleSubMenuDismiss = (event, documentClick) => {
-    if ((event && event.keyCode === keycode.codes.tab) || documentClick) {
+  handleSubMenuDismiss = (
+    event: React.MouseEvent | React.KeyboardEvent,
+    documentClick: boolean
+  ) => {
+    if (
+      (event && (event as React.KeyboardEvent).keyCode === keycode.codes.tab) ||
+      documentClick
+    ) {
       this.hide(event)
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  hide = (event) => {
+  hide = (event: React.MouseEvent | React.KeyboardEvent) => {
     if (this._popover) {
-      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-      this._popover.hide(event)
+      this._popover.hide(event as unknown as Event)
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  show = (event) => {
+  show = (event?: React.MouseEvent | React.KeyboardEvent) => {
     if (this._popover) {
-      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-      this._popover.show(event)
+      //TODO make events consistents
+      this._popover.show(event as unknown as Event)
     }
   }
 
   focus() {
     if (this.shown) {
-      // @ts-expect-error ts-migrate(2555) FIXME: Expected at least 5 arguments, but got 2.
-      error(this._menu && this._menu.focus, '[Menu] Could not focus the menu.')
-      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-      this._menu.focus()
+      const hasFocusOnMenu =
+        this._menu !== null && typeof this._menu.focus === 'function'
+
+      error(hasFocusOnMenu, '[Menu] Could not focus the menu.')
+
+      if (hasFocusOnMenu) {
+        this._menu!.focus()
+      }
+      if (this._trigger !== null && typeof this._trigger.focus === 'function') {
+        this._trigger.focus()
+      }
```

Comment:
Imho this if-else block got a bit overcomplicated in the back-and-forth refactoring, and it is hard to tell wether if it is still equivalent to the original. This method basically tries to focus the menu, and if it cannot, throws an error and tries to focus the trigger instead, and a warning if that is not possible either. I suggest writing it like this, seems clean and simple and is equal to the original:

```js
  focus() {
    if (this.shown) {
      error(!!this._menu?.focus, '[Menu] Could not focus the menu.')
      this._menu!.focus()
    } else {
      error(!!this._trigger?.focus, '[Menu] Could not focus the trigger.')
      this._trigger!.focus!()
    }
  }
```

Reply:
The other checks are more secure, but probably won't be an issue either way. 

I'll replace the block with this, just to be more readable